### PR TITLE
Set compiler libraries to netstandard2.0"

### PR DESCRIPTION
**Pull request by KevinRansom**
FSharp.Compiler.Private.dll and FSharp.Compiler.Interactive.Settings.dll still built against netstandard1.6.

Updating to 2.0.  Since we never deploy recent builds on netcoreapp1.* any longer.

### CODE REVIEW

**Comment by dsyme** [comment]
Yay!

**Comment by KevinRansom** [comment]
Lol

**Comment by cartermp** [comment]
Just to note this may break vs2017 nightlies if merged

**Comment by KevinRansom** [comment]
@cartermp, I think it should be okay, even in master with 15.9.

**Comment by Pilchie** [comment]
Honestly, given that RC is out, I'd be fine if the nightlies only work on 16.0 anyway.

**Comment by Pilchie** [comment]
(Wrong button, sorry @KevinRansom)

**Comment by dsyme** [comment]
> Honestly, given that RC is out, I'd be fine if the nightlies only work on 16.0 anyway.

So far we've used the policy that the nightlies should work with the latest non-preview release of VS.  I think it's best to stick to that for consistency?

File: src/fsharp/ExtensionTyping.fs [hunk 1]
```diff
@@ -371,20 +367,11 @@ module internal ExtensionTyping =
...
-#if FX_RESHAPED_REFLECTION
-        inherit ProvidedMemberInfo(x.GetTypeInfo(), ctxt)
-#if FX_NO_CUSTOMATTRIBUTEDATA
-        let provide () = ProvidedCustomAttributeProvider.Create (fun provider -> provider.GetMemberCustomAttributesData(x.GetTypeInfo()) :> _)
-#else
-        let provide () = ProvidedCustomAttributeProvider.Create (fun _provider -> x.GetTypeInfo().CustomAttributes)
-#endif
-#else
         inherit ProvidedMemberInfo(x, ctxt)
 #if FX_NO_CUSTOMATTRIBUTEDATA
```
Comment (dsyme) [inline src/fsharp/ExtensionTyping.fs]: `FX_NO_CUSTOMATTRIBUTEDATA` is always off, we can remove it too :)

**Comment by Pilchie** [comment]
> So far we've used the policy that the nightlies should work with the latest non-preview release of VS. I think it's best to stick to that for consistency?

I agree in principle, but given that RC is go live, and in the Release channel, if it became an issue, I think we could say that it meets the spirit of the law.  I wouldn't suggest this for a regular Preview.  Anyway, it sounds like it's not an issue according to @KevinRansom.  If it is, the other option would just be to merge into the dev16 branch and wait until release comes out to merge to master.

**Comment by KevinRansom** [comment]
It shouldn't be a problem.  We can easily build a nightly from this PR and verify that everything works as we expect.

**Comment by cartermp** [comment]
@KevinRansom are we targeting 16.1 for this?

**Comment by TIHan** [comment]
This is looking good! 👍

**Comment by KevinRansom** [comment]
@cartermp 

I was thinking get it into master and then 16.1

**Comment by KevinRansom** [comment]
@cartermp we are pretty much done with 16.0

**Pull request merged**